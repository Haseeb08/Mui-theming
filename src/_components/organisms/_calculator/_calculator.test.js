import React from "react";
import { create } from "react-test-renderer";
import { render,fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Calculator from "./_calculator";

describe("Calculator testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<Calculator />);
    expect(input.toJSON()).toMatchSnapshot();
  });
  test("should have calculator", () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId("calculator")).toBeInTheDocument();
  });
  test("should have title", () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId("title")).toBeInTheDocument();
  });
  test("should have input", () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId("input")).toBeInTheDocument();
  });
  test("should have keypad", () => {
    const { getByTestId } = render(<Calculator />);
    expect(getByTestId("keypad")).toBeInTheDocument();
  });
  test("should have all buttons buttons", () => {
    const { getAllByTestId } = render(<Calculator />);
    expect(getAllByTestId("button")).toHaveLength(18);
  });

  test("Perform Calculation", () => {
    const { getAllByTestId,getByTestId } = render(<Calculator />);
    const buttons = getAllByTestId("button");
    const input = getByTestId("input");
    expect(input).toHaveTextContent("");
    fireEvent.click(buttons[7]);
    expect(input).toHaveTextContent("5");
    fireEvent.click(buttons[5]);
    expect(input).toHaveTextContent("5+");
    fireEvent.click(buttons[6]);
    expect(input).toHaveTextContent("5+6");
    fireEvent.click(buttons[16]);
    expect(input).toHaveTextContent("11");
    fireEvent.click(buttons[1]);
    expect(input).toHaveTextContent("1");
    fireEvent.click(buttons[5])
    fireEvent.click(buttons[5])
    fireEvent.click(buttons[16]);
    expect(input).toHaveTextContent("error");
    fireEvent.click(buttons[0]);
    expect(input).toHaveTextContent("");

  });
});
