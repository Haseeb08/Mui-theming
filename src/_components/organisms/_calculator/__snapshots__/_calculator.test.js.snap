// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator testing Matches the snapshot 1`] = `
<div
  className="makeStyles-calcWrapper-1"
  data-testid="calculator"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
    data-testid="title"
  >
    MUI Calculator
  </h3>
  <hr
    style={
      Object {
        "margin": 0,
      }
    }
  />
  <div
    className="makeStyles-input-2"
    data-testid="input"
  >
    
  </div>
  <hr
    style={
      Object {
        "margin": 0,
      }
    }
  />
  <div
    className="makeStyles-keyPad-3"
    data-testid="keypad"
  >
    <div
      className="makeStyles-keyRow-4"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-largeButton-6 [object Object]"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              AC
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-largeButton-6 [object Object]"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              &lt;=
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      className="makeStyles-keyRow-4"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              9
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              8
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              7
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedSecondary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              +
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      className="makeStyles-keyRow-4"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              6
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              5
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              4
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedSecondary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              -
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      className="makeStyles-keyRow-4"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              3
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              2
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              1
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedSecondary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              *
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      className="makeStyles-keyRow-4"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              0
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedPrimary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              .
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 makeStyles-equalButton-7"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              =
            </span>
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 [object Object] MuiButton-containedSecondary"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
               
              /
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
